### PR TITLE
BFS Article

### LINKED ISSUE #63: Article for BFS
## 📚 Documentation

(A clear and concise description of what the issue is.)

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

How BFS works and Implementation of it in c#

### PR DESCRIPTION
**Pull request by SalmaZakaria**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

An article explaining BFS Algorithm

### Checklist

- [Yes] I've read the contribution guidelines.
- [Yes] I've checked the issue list before deciding what to submit.
- [Yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #63

### CODE REVIEW

**Comment by vybhav72954** [comment]
Link the issue properly

**Comment by SalmaZakaria** [comment]
@vybhav72954  I have already linked the issue no. 63

File: docs/graph-algorithms/breadth-first-search.mdx [hunk 1]
```diff
@@ -0,0 +1,101 @@
+---
+title: Breadth First Search
+slug: /graph-algorithms/breadth-first-search
+---
+
+## Introduction
+Breadth First Search Algorithm (BFS) is a Popular algorith to find shortest path in unwighted graph
```
Comment (HarshCasper) [inline docs/graph-algorithms/breadth-first-search.mdx:7]: This does not explain anything. Spell mistakes here and there. I would suggest a rewrite of this portion.

File: docs/graph-algorithms/breadth-first-search.mdx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+## Explanation
+
+- Given an undirected & Connected Graph (V, E)
+  V: Vertices E: Edges
+- Goal: methodically explore every vertex and every edge
+- “Explore” a graph, turning it into a tree
+- One vertex at a time
+- Expand frontier of explored vertices across its breadth
+- Discovers all vertices at distance k from s before those at distance k+1
+- A level-by-level order traversal
```
Comment (HarshCasper) [inline docs/graph-algorithms/breadth-first-search.mdx:18]: It doesn't explain anything as well. Prefer using diagrams and representations to explain the same.

File: docs/graph-algorithms/breadth-first-search.mdx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+##APPS
```
Comment (HarshCasper) [inline docs/graph-algorithms/breadth-first-search.mdx:20]: What does `APPS` mean?

File: docs/graph-algorithms/breadth-first-search.mdx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        Color[it->first] = 'G';
+        Q.push(it->first)
+	BFS(Q, Boxes, Parents, Color);
+	return 0;
+}
+
+```
+
+## Complexity
+- Time Complexity: O(V+E)
```
Comment (HarshCasper) [inline docs/graph-algorithms/breadth-first-search.mdx:100]: Format the values.

File: docs/graph-algorithms/breadth-first-search.mdx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
```
Comment (HarshCasper) [inline docs/graph-algorithms/breadth-first-search.mdx:101]: Add the Credits

File: docs/graph-algorithms/breadth-first-search.mdx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+- All vertices start out white
+- Grey vertices are discovered but not fully explored
+- They may be adjacent to white vertices
+- Black vertices are discovered and fully explored
+- They are adjacent only to black and gray vertices
+- Explore vertices by scanning neighbors of grey vertices
+
+## Code
+
+```cpp
```
Comment (HarshCasper) [inline docs/graph-algorithms/breadth-first-search.mdx:46]: Add more implementations. See some merged PRs to see how Code is actually added here

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/graph-algorithms/breadth-first-search.mdx (hunk 5 above)
Comment (SalmaZakaria) [reply docs/graph-algorithms/breadth-first-search.mdx:101]: Credits to the code?
I write it by myself

File: docs/graph-algorithms/breadth-first-search.mdx (hunk 2 above)
Comment (SalmaZakaria) [reply docs/graph-algorithms/breadth-first-search.mdx:18]: @HarshCasper  Can I add images in the doc?

File: docs/graph-algorithms/breadth-first-search.mdx (hunk 6 above)
Comment (SalmaZakaria) [reply docs/graph-algorithms/breadth-first-search.mdx:50]: @HarshCasper Changes are done

**Pull request closed without merging**